src/main/java/io/odpf/depot/bigquery/json/BigqueryJsonUpdateListener.java

```diff
@@ -1,24 +1,80 @@
 package io.odpf.depot.bigquery.json;
 
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.FieldList;
+import com.google.cloud.bigquery.LegacySQLTypeName;
+import com.google.cloud.bigquery.Schema;
+import io.odpf.depot.bigquery.handler.BigQueryClient;
 import io.odpf.depot.bigquery.handler.MessageRecordConverter;
 import io.odpf.depot.bigquery.handler.MessageRecordConverterCache;
+import io.odpf.depot.common.TupleString;
 import io.odpf.depot.config.BigQuerySinkConfig;
 import io.odpf.depot.message.OdpfMessageParser;
 import io.odpf.depot.stencil.OdpfStencilUpdateListener;
 
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
 public class BigqueryJsonUpdateListener extends OdpfStencilUpdateListener {
     private final MessageRecordConverterCache converterCache;
     private BigQuerySinkConfig config;
+    private final BigQueryClient bigQueryClient;
 
-    public BigqueryJsonUpdateListener(BigQuerySinkConfig config, MessageRecordConverterCache converterCache) {
+    public BigqueryJsonUpdateListener(BigQuerySinkConfig config, MessageRecordConverterCache converterCache, BigQueryClient bigQueryClient) {
         this.converterCache = converterCache;
         this.config = config;
+        this.bigQueryClient = bigQueryClient;
+        if (!config.getSinkConnectorSchemaJsonDynamicSchemaEnable()) {
+            throw new UnsupportedOperationException("currently only schema inferred from incoming data is supported, stencil schema support for json will be added in future");
+        }
     }
 
     @Override
     public void updateSchema() {
         OdpfMessageParser parser = getOdpfMessageParser();
         MessageRecordConverter messageRecordConverter = new MessageRecordConverter(parser, config, null);
         converterCache.setMessageRecordConverter(messageRecordConverter);
+        Schema existingTableSchema = bigQueryClient.getSchema();
+        FieldList existingTableFields = existingTableSchema.getFields();
+        List<TupleString> defaultColumns = config.getSinkBigquerySchemaJsonOutputDefaultColumns();
+        Set<Field> defaultFields = defaultColumns
+                .stream()
+                .map(tupleString -> getField(tupleString))
+                .collect(Collectors.toSet());
+        Set<Field> finalFieldList = new HashSet<Field>(defaultFields);
```

Comment:
`
Set<Field> finalFieldList = defaultColumns.stream().map(this::getField).collect(Collectors.toSet());
`

Reply:
Excerpt from javadoc of [Collectors.toSet](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#toSet--)
"Returns a Collector that accumulates the input elements into a new Set. There are no guarantees on the type, mutability, serializability, or thread-safety of the Set returned; if more control over the returned Set is required, use [toCollection(Supplier)](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#toCollection-java.util.function.Supplier-)." 
The javadoc says the returned set immutability is not guaranteed, to have control over what type of set will be returned use Collectors.toCollection. 
So based on the suggestion in the javadoc will update the code to:
`.collect(Collectors.toCollection(HashSet::new));`

Reply:
alright. makes sense. just look at the another comment that I made.. small thing